service/evaluation/evaluation.go

```diff
@@ -0,0 +1,429 @@
+		err = status.Errorf(codes.AlreadyExists, "Cloud Service '%s' is being evaluated with Control %s already.", req.Toe.CloudServiceId, req.ControlId)
+		log.Error(err)
+		return
+	}
+	s.evaluationMutex.Unlock()
+
+	log.Info("Starting evaluation ...")
+	s.scheduler.TagsUnique()
+
+	log.Infof("Evaluate Cloud Service '%s' for Control ID '%s' every 5 minutes...", req.Toe.CloudServiceId, req.ControlId)
+	_, err = s.scheduler.
+		Every(5).
+		Minute().
+		Tag(schedulerTag).
+		Do(s.Evaluate, req)
+	if err != nil {
+		err = fmt.Errorf("evaluation for Cloud Service '%s' and Control ID '%s' cannot be scheduled", req.Toe.CloudServiceId, req.ControlId)
+		log.Error(err)
+		err = status.Errorf(codes.Internal, "%s", err)
+		log.Error(err)
+		return
+	}
+
+	// Add map entry for target cloud service id and control id
+	s.evaluationMutex.Lock()
+	if s.evaluation[schedulerTag] == nil {
+		s.evaluation[schedulerTag] = new(EvaluationScheduler)
+		s.evaluation[schedulerTag].scheduler = s.scheduler
+		s.evaluation[schedulerTag].evaluatedControlID = req.ControlId
+	}
+	s.evaluationMutex.Unlock()
+
+	s.scheduler.StartAsync()
+
+	return
+}
+
+func (s *Service) Evaluate(req *evaluation.StartEvaluationRequest) {
+	log.Infof("Started evaluation for Cloud Service '%s',  Catalog ID '%s' and Control '%s'", req.Toe.CloudServiceId, req.Toe.CatalogId, req.ControlId)
+
+	// Get orchestrator client
+	err := s.initOrchestratorClient()
+	if err != nil {
+		log.Errorf("could not set orchestrator client: %v", err)
+		return
+	}
+	// Get control and the associated metrics
+	metrics, err := s.getMetricsFromControl(req)
+	if err != nil {
+		log.Errorf("Could not get metrics for control ID '%s' from Orchestrator: %v", req.ControlId, err)
+
+		// TODO(anatheka): Do we need that? Or do we let it running?
+		// Delete evaluation entry, it is not longer needed
+		s.evaluationMutex.Lock()
+		delete(s.evaluation, createSchedulerTag(req.Toe.CloudServiceId, req.ControlId))
+		s.evaluationMutex.Unlock()
+
+		return
+	}
+
+	// Get assessment results for the target cloud service
+	assessmentResults, err := api.ListAllPaginated(&assessment.ListAssessmentResultsRequest{FilteredCloudServiceId: req.Toe.CloudServiceId}, s.orchestratorClient.ListAssessmentResults, func(res *assessment.ListAssessmentResultsResponse) []*assessment.AssessmentResult {
+		return res.Results
+	})
+	if err != nil {
+		log.Errorf("Could not get assessment results for Cloud Serivce '%s' from Orchestrator", req.Toe.CloudServiceId)
+
+		// TODO(anatheka): Do we need that? Or do we let it running?
+		// Delete evaluation entry, it is no longer needed if we don't get the assessment results from the orchestrator
+		s.evaluationMutex.Lock()
+		delete(s.evaluation, createSchedulerTag(req.Toe.CloudServiceId, req.ControlId))
+		s.evaluationMutex.Unlock()
+		return
+	}
+
+	// Get mapping assessment results related to the metric
+	mappingList := getMapping(assessmentResults, metrics)
```

Comment:
I would extract this into the `ListAssessmentResults` call of the orchestrator.

Reply:
@oxisto should I start a PR for that or do you have something where you can add it?